src/parser.rs

```diff
@@ -56,13 +56,11 @@ where
 parser! {
     type PartialState = AnySendPartialState;
     fn value['a, I]()(I) -> RedisResult<Value>
-        where [I: RangeStream<Item = u8, Range = &'a [u8]> ]
+        where [I: FullRangeStream<Item = u8, Range = &'a [u8]> ]
     {
-        let end_of_line: fn () -> _ = || crlf().or(newline());
-        let line = || recognize(take_until_range(&b"\r\n"[..]).with(end_of_line()))
+        let line = || recognize(take_until_bytes(&b"\r\n"[..]).with(take(2).map(|_| ())))
```

Comment:
Just for my understanding (haven't used combine myself at all yet): The `\r\n` will be included in the final line and thus we can safely assume `line.len() below is at least 2, so the `-2` is safe?

Reply:
Exactly, `recognize` returns the entire range (`&[u8]`) covered by what its argument has parsed. Since the argument parses everything up until `\r\n` and then takes two more elements (which is the actual `\r\n` bytes) it contains at least two bytes.

Reply:
Ah, that explains the "with" then. All good to go!